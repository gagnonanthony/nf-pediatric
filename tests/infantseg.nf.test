nextflow_pipeline {

    options "-stub-run"

    name "Test nf-pediatric profile segmentation"
    script "../main.nf"

    test("Segmentation profile - MCRIBS Infant - should run successfully") {

        when {
            params {

                params.input = "$projectDir/tests/data/bids-infant"
                params.outdir = "$outputDir"
                params.fs_output_dir = "$outputDir/../onedirdown/mcribs-2.1.0"

                params.segmentation = true
                params.fs_license = "https://www.dropbox.com/scl/fi/0s8lp6lydyd0rxawxb4jm/license.txt?rlkey=hz54oc0d4sor69avqphtrjvgn&st=9e0yij97&dl=0"

                params.infant = true
                params.t1_run_synthstrip = true
                params.t2_run_synthstrip = true

            }
        }

        then {
            // stable name: All files + folders in ${params.outdir}/ with a stable name.
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore:  ['pipeline_info/*.{html,json,txt}'])
            // All mcribs output.
            def mcribs_output = getAllFilesFromDir(params.fs_output_dir, relative: true, includeDir: true)
            assertAll {
                assert workflow.success
                assert snapshot(
                    // Number of successfully completed tasks
                    workflow.trace.succeeded().size(),
                    // Remove the nextflow version from the versions.yml because we test it using different nextflow versions.
                    removeNextflowVersion("$outputDir/pipeline_info/pipeline_software_mqc_versions.yml"),
                    // All stable name.
                    stable_name,
                    // All mcribs output.
                    mcribs_output
                ).match()
            }
        }
    }
}
