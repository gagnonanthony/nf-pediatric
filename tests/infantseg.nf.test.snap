{
    "Segmentation profile - MCRIBS Infant - should run successfully": {
        "content": [
            20,
            {
                "DENOISING_NLMEANS": {
                    "scilpy": "2.0.2"
                },
                "Workflow": {
                    "scilus/nf-pediatric": "v1.0dev"
                }
            },
            [
                "cortical_area_lh.tsv",
                "cortical_area_rh.tsv",
                "cortical_thickness_lh.tsv",
                "cortical_thickness_rh.tsv",
                "cortical_volume_lh.tsv",
                "cortical_volume_rh.tsv",
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf-pediatric_software_mqc_versions.yml",
                "sub-test",
                "sub-test/ses-baseline",
                "sub-test/ses-baseline/anat",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_desc-boundingbox.pkl",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_desc-brain_mask.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_desc-preproc_T1w.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_desc-preproc_T2w.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_from-T1w_to-T2w_affine.mat",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_from-T2w_to-T1w_affine.mat",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_seg-DK_dseg.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_seg-DK_stat-lh_area.tsv",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_seg-DK_stat-lh_thickness.tsv",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_seg-DK_stat-lh_volume.tsv",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_seg-DK_stat-rh_area.tsv",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_seg-DK_stat-rh_thickness.tsv",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_seg-DK_stat-rh_volume.tsv",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_seg-DK_stat-subcortical.tsv",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_space-T2w_desc-preproc_T1w.nii.gz",
                "sub-test/ses-baseline/figures",
                "sub-test/ses-baseline/figures/sub-test_labels_mqc.png",
                "sub-test/ses-baseline/multiqc",
                "sub-test/ses-baseline/multiqc/sub-test_multiqc_data",
                "sub-test/ses-baseline/multiqc/sub-test_multiqc_plots",
                "sub-test/ses-baseline/multiqc/sub-test_multiqc_report.html",
                "subcortical_volumes.tsv"
            ],
            [
                "",
                "sub-test_${meta.session}"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-02-07T12:31:17.066027"
    }
}